plugins/woocommerce/src/Blocks/BlockTypes/ProductButton.php

```diff
@@ -178,31 +177,22 @@ protected function render( $attributes, $content, $block ) {
 			$this->prevent_cache();
 		}
 
-		$interactive = array(
-			'namespace' => 'woocommerce/product-button',
-		);
-
 		$div_directives = '
-			data-wc-interactive=\'' . wp_json_encode( $interactive, JSON_NUMERIC_CHECK | JSON_HEX_TAG | JSON_HEX_APOS | JSON_HEX_QUOT | JSON_HEX_AMP ) . '\'
-			data-wc-context=\'' . wp_json_encode( $context, JSON_NUMERIC_CHECK | JSON_HEX_TAG | JSON_HEX_APOS | JSON_HEX_QUOT | JSON_HEX_AMP ) . '\'
+			data-wp-interactive="woocommerce/product-button"
+			data-wp-context=\'' . wp_json_encode( $context, JSON_NUMERIC_CHECK | JSON_HEX_TAG | JSON_HEX_APOS | JSON_HEX_QUOT | JSON_HEX_AMP ) . '\'
+			data-wp-init="actions.refreshCartItems"
 		';
 
-		$button_directives = '
-			data-wc-on--click="actions.addToCart"
-			data-wc-class--loading="context.isLoading"
-		';
-
-		$anchor_directive = '
-			data-wc-on--click="woocommerce/product-collection::actions.viewProduct"
-		';
+		$button_directives = 'data-wp-on--click="actions.addCartItem"';
+		$anchor_directive  = 'data-wp-on--click="woocommerce/product-collection::actions.viewProduct"';
```

Comment:
Is this directive also injected [from here](https://github.com/woocommerce/woocommerce/blob/04c998dc520c8112288c4babb67a6adfb88146bc/plugins/woocommerce/src/Blocks/BlockTypes/ProductCollection/Controller.php#L155), so maybe we are repeating it?